src/Tribe/Views/V2/iCalendar/Traits/Outlook_Methods.php

```diff
@@ -44,33 +45,31 @@ trait Outlook_Methods {
 	 * Generate the parameters for the Outlook export buttons.
 	 *
 	 * @since 5.16.0
+	 * @since TBD Adding timezone to the start and end dates generated.
 	 *
 	 * @param string $calendar Whether it's Outlook live or Outlook 365.
 	 *
 	 * @return string Part of the URL containing the event information.
 	 */
 	protected function generate_outlook_add_url_parameters( $calendar = 'live' ) {
 		// Getting the event details
-		$event = tribe_get_event();
-
-		$path = '/calendar/action/compose';
-		$rrv  = 'addevent';
+		$event    = tribe_get_event();
+		$path     = '/calendar/action/compose';
+		$rrv      = 'addevent';
+		$timezone = $event->timezone ?? Tribe__Timezones::wp_timezone_string();
 
 		/**
 		 * If event is an all day event, then adjust the end time.
 		 * Using the 'allday' parameter doesn't work well through time zones.
 		 */
 		if ( $event->all_day ) {
-			$enddt = Dates::build_date_object( $event->end_date )->format( 'Y-m-d' )
-				. 'T'
-				. Dates::build_date_object( $event->start_date )->format( 'H:i:s' );
+			$enddt = Dates::build_date_object( $event->end_date, $timezone )->format( 'Y-m-d' ) . 'T'
+			         . Dates::build_date_object( $event->start_date, $timezone )->format( 'H:i:s' );
 		} else {
-			$enddt = Dates::build_date_object( $event->end_date )->format( 'c' );
-			$enddt = substr( $enddt, 0, strlen( $enddt ) - 6 );
+			$enddt = Dates::build_date_object( $event->end_date, $timezone )->format( 'c' );
 		}
```

Comment:
Shouldn't we have some check to make sure the `$event` is a valid event, otherwise if `null` is passed from `tribe_get_event` that can throw fatals.